File: Python/Algorithms/2315.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        # True when we encounter first '|' false when encounter second.
        count, flag = 0, False
        for char in s:
            if char == '|':
                flag = not flag
                continue
            if char == '*' and not flag:
                count += 1
        return count

File: C/Algorithms/2315.c
int countAsterisks(char * s){
    int count = 0, flag = 0;
    char c;
    while (c = *s++) {
        if (c == '|') {
            flag ^= 1;
            continue;
        }
        if (c == '*' && !flag)
            count++;
    }
    
    return count;
}